src/main/java/de/tum/in/www1/artemis/service/CourseScoreCalculationService.java

```diff
@@ -98,35 +126,161 @@ public CourseScoresDTO calculateCourseScores(long courseId, Collection<Long> stu
+        PlagiarismVerdict mostSeverePlagiarismVerdict = null;
+        boolean hasParticipated;
+        PlagiarismMapping plagiarismMapping = PlagiarismMapping.createFromPlagiarismCases(plagiarismCases);
+        if (entry.getValue().isEmpty()) {
+            presentationScorePassed = false;
+            hasParticipated = false;
+        }
+        else if (plagiarismMapping.studentHasVerdict(entry.getKey(), PlagiarismVerdict.PLAGIARISM)) {
+            presentationScorePassed = false;
+            mostSeverePlagiarismVerdict = PlagiarismVerdict.PLAGIARISM;
+            hasParticipated = true;
+        }
+        else {
+            presentationScorePassed = isPresentationScoreSufficientForBonus(studentScores.getPresentationScore(), course.getPresentationScore());
+            Map<Long, PlagiarismCase> plagiarismCasesForStudent = plagiarismMapping.getPlagiarismCasesForStudent(entry.getKey());
+            mostSeverePlagiarismVerdict = findMostServerePlagiarismVerdict(plagiarismCasesForStudent.values());
+            hasParticipated = true;
+        }
+        return new StudentScoresForExamBonusSourceDTO(studentScores.getAbsoluteScore(), studentScores.getRelativeScore(), studentScores.getCurrentRelativeScore(),
+                studentScores.getPresentationScore(), entry.getKey(), presentationScorePassed, mostSeverePlagiarismVerdict, hasParticipated);
+    }
+
+    /**
+     * Get all the items needed for the CourseForDashboardDTO.
+     * This includes scoresPerExerciseType and participationResults.
+     *
+     * @param course the course to calculate the items for.
+     * @param userId the id of the students whose scores in the course will be calculated.
+     * @return the CourseForDashboardDTO containing all the mentioned items.
+     */
+    public CourseForDashboardDTO getScoresAndParticipationResults(Course course, long userId) {
+        List<StudentParticipation> studentParticipations = new ArrayList<>();
+        for (Exercise exercise : course.getExercises()) {
+            exercise.setCourse(course);
+            StudentParticipation exerciseParticipation;
+            // This method is used in the CourseResource where the course is first fetched with lazy participations, and participations are then fetched separately in the
+            // CourseService
+            // and added to the course if found.
+            // If no participations are found for the course, no value is set to the course's participations and trying to access them here would throw a
+            // LazyInitializationException. This is why we first need to check if the participations are initialized before adding them to the list of participations.
+            // If they are not initialized, this means that there are no participations for this exercise and user.
+            if (Hibernate.isInitialized(exercise.getStudentParticipations())) {
+                exerciseParticipation = exercise.getStudentParticipations().iterator().next();
```

Comment:
Can you be sure that there only is one participation and that this is also always the correct one? What about the practice mode, here a student could have a graded and ungraded participation.

Reply:
Yes. The participations here are coming from `StudentParticipationRepository: findByStudentIdAndIndividualExercisesWithEagerSubmissionsResultIgnoreTestRuns`